src/websockets/stream_handler.js

```diff
@@ -136,31 +128,28 @@ define(["websockets/binary_websockets", "charts/chartingRequestMap", "common/uti
 
                 var type = $(chartID.containerIDWithHash).data('type');
 
-                var last = series.data[series.data.length - 1];
-                if (series.options.data.length != series.data.length) {
-                    //TODO - This is an error situation
-                    setExtremePointsForXAxis(chart, null, bar.time);
-                    return;
-                }
+                var len = chart.series[0].options.data.length - 1,
+                    last = chart.series[0].options.data[len];
 
                 if (type && isDataTypeClosePriceOnly(type)) {//Only update when its not in loading mode
                     if (isNew) {
                         series.addPoint([time, close], true, true, false);
                     } else {
-                        last.update({
-                            y : close
-                        });
+                        last[1] = close;
+                        chart.series[0].options.data[len] = last;
+                        // Redrawing the chart to update new point.
+                        chart.series[0].setData(chart.series[0].options.data, true);
```

Comment:
can you send me the link. This seems to be very bad suggestion. For every update, it is going to reset so many things and refresh all data.